src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1318,9 +1308,24 @@ void RaftReplDev::on_fetch_data_received(intrusive< sisl::GenericRpcData >& rpc_
     folly::collectAllUnsafe(futs).thenValue(
         [this, rpc_data = std::move(rpc_data), sgs_vec = std::move(sgs_vec)](auto&& vf) {
             for (auto const& err_c : vf) {
-                if (sisl_unlikely(err_c.value())) {
+                RD_REL_ASSERT(err_c.hasValue(), "should have an error code in fetching data");
+                const auto& err = err_c.value();
+                if (sisl_unlikely(err)) {
                     COUNTER_INCREMENT(m_metrics, read_err_cnt, 1);
-                    RD_REL_ASSERT(false, "Error in reading data");
```

Comment:
have we ever hit here? Yawen and I did the test before and as the comment indicate it fails in IOMgr.

And in current implementation if we return an empty image, wont the client (folllower) panic?

```
void RaftReplDev::handle_fetch_data_response(sisl::GenericClientResponse response,
                                             std::vector< repl_req_ptr_t > rreqs) {
    auto resp_blob = response.response_blob();
    auto raw_data = resp_blob.cbytes();
    auto total_size = resp_blob.size();

    COUNTER_INCREMENT(m_metrics, fetch_total_blk_size, total_size);

    RD_DBG_ASSERT_GT(total_size, 0, "Empty response from remote");
    RD_DBG_ASSERT(raw_data, "Empty response from remote");

    RD_LOGD(NO_TRACE_ID, "Data Channel: FetchData completed for {} requests", rreqs.size());
```

Reply:
yes, I hit here once, but only once.

```
RD_DBG_ASSERT_GT(total_size, 0, "Empty response from remote"); RD_DBG_ASSERT(raw_data, "Empty response from remote");
```
yes , this will make follower panic if empty image received.  there two are very old assert, which was made one year ago. we can remove it add return directly if got empty blob?


theoretically，we might encounter read error [here](https://github.com/eBay/HomeObject/blob/1923a585edaa12a3147dbf127db61a5e337ebf11/src/lib/homestore_backend/replication_state_machine.cpp#L764).